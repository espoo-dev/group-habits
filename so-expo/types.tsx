/**
 * Learn more about using TypeScript with React Navigation:
 * https://reactnavigation.org/docs/typescript/
 */

import { NavigatorScreenParams } from '@react-navigation/native';
import { NativeStackScreenProps } from '@react-navigation/native-stack';

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList { }
  }
}

export type RootStackParamList = {
  Root: NavigatorScreenParams<RootParamList> | undefined;
  SignIn: undefined;
  SignUp: undefined;
  Modal: undefined;
  NotFound: undefined;
};

export type PrivateStackParamList = {
  Dashboard: NavigatorScreenParams<RootParamList> | undefined;
  Modal: undefined;
  NotFound: undefined;
};

export type RootParamList = {
  Home: undefined;
};

export type RootStackScreenProps<Screen extends keyof RootStackParamList> = NativeStackScreenProps<
  RootStackParamList,
  Screen
>;
