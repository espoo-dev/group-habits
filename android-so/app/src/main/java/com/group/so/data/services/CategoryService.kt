package com.group.so.data.services

import com.group.so.data.entities.network.CategoryDTO
import com.group.so.data.entities.request.CategoryDataRequest
import okhttp3.ResponseBody
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface CategoryService {
    @GET("categories")
    suspend fun getAllCategories(): List<CategoryDTO>

    @POST("categories")
    suspend fun registerCategory(@Body categoryDataRequest: CategoryDataRequest): CategoryDTO

    @DELETE("categories/{id}")
    suspend fun deleteCategory(@Path("id") id: Int): Response<ResponseBody>

    @PUT("categories")
    suspend fun editCategory(@Body categoryDataRequest: CategoryDataRequest): CategoryDTO
}
