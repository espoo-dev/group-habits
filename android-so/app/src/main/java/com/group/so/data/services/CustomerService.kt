package com.group.so.data.services

import com.group.so.data.entities.network.CustomerDTO
import com.group.so.data.entities.request.customer.CustomerDataRequest
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path
import retrofit2.http.Query

interface CustomerService {

    @POST("customers")
    suspend fun registerCustomer(@Body customerDataRequest: CustomerDataRequest): CustomerDTO

    @PUT("customers/{id}")
    suspend fun editCustomer(
        @Path("id") id: Int,
        @Body customerDataRequest: CustomerDataRequest
    ): CustomerDTO

    @GET("customers")
    suspend fun getAllCustomers(): List<CustomerDTO>

    @GET("customers")
    suspend fun getCustomersByName(@Query("name") name: String?): List<CustomerDTO>

    @GET("customers")
    suspend fun getCustomersByCustomerType(
        @Query("customer_type") customType: String?
    ): List<CustomerDTO>
}
