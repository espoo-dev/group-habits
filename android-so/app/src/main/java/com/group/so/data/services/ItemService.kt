package com.group.so.data.services

import com.group.so.data.entities.network.ItemDTO
import com.group.so.data.entities.request.product.ProductDataRequest
import com.group.so.data.entities.request.service.ServiceDataRequest
import okhttp3.ResponseBody
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path
import retrofit2.http.Query

interface ItemService {

    @POST("items")
    suspend fun registerService(@Body serviceDataRequest: ServiceDataRequest): ItemDTO

    @POST("items")
    suspend fun registerProduct(@Body productDataRequest: ProductDataRequest): ItemDTO

    @GET("items")
    suspend fun getAllItems(): List<ItemDTO>

    @GET("items")
    suspend fun getItemByName(@Query("name") name: String?): List<ItemDTO>

    @GET("items")
    suspend fun getItemsByItemType(
        @Query("item_type") customType: String?
    ): List<ItemDTO>

    @GET("items")
    suspend fun getItemsByNameAndType(
        @Query("name") name: String?,
        @Query("item_type") customType: String?
    ): List<ItemDTO>

    @DELETE("items/{id}")
    suspend fun deleteItem(@Path("id") id: Int): Response<ResponseBody>

    @PUT("items/{id}")
    suspend fun editService(
        @Path("id") id: Int,
        @Body serviceDataRequest: ServiceDataRequest
    ): ItemDTO

    @PUT("items/{id}")
    suspend fun editProduct(
        @Path("id") id: Int,
        @Body productRequest: ProductDataRequest
    ): ItemDTO
}
