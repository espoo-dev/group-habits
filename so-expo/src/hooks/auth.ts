import { useContext } from "react";
import { AuthContext } from "../contexts/AuthContext";

function useAuth() {
  const { auth } = useContext(AuthContext)

  return auth;
}

export { useAuth };
