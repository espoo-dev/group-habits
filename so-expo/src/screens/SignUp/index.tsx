import { StyleSheet } from "react-native";
import { Text, View } from "../../components/Themed";

function SignUp() {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>
        Login
      </Text>

      <View style={styles.card}>
        <Text>
          Card aqui
        </Text>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  card: {
    backgroundColor: 'red',
    borderRadius: 8,
    marginTop: 10,
    padding: 18,
  },
})

export { SignUp };
