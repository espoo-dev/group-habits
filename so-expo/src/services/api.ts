import { REACT_APP_API_URL } from "@env";
import axios from "axios";

const headers = {
  "Accept": "application/json",
  "Content-Type": "application/json",
}

const api = axios.create({
  baseURL: REACT_APP_API_URL,
  headers,
})

export { api };
