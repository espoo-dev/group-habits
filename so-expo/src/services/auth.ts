import { api } from "./api";

type UserCredentials = {
  email: string;
  password: string;
}

type AuthParams = {
  user: UserCredentials;
}

const auth = async (params: AuthParams) => {
  try {
    const { data } = await api.post('/v1/users/sign_in', params)
    return data;
  } catch (error) {
    console.error('error', error)
  }
}

export { auth };
