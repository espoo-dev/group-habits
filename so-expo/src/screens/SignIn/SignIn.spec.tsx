import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { fireEvent, render, screen } from '@testing-library/react-native';
import { RootStackParamList } from '../../../types';
import * as authService from "../../services/auth";
import { SignUp } from '../SignUp';
import { SignIn } from './index';

const Stack = createNativeStackNavigator<RootStackParamList>();

const component = (
  <NavigationContainer>
    <Stack.Navigator>
      <Stack.Screen
        name="SignIn"
        component={SignIn}
      />
      <Stack.Screen
        name="SignUp"
        component={SignUp}
      />
    </Stack.Navigator>
  </NavigationContainer>
);

const setup = () => {
  return render(component)
}

describe('src > screens > SignIn', () => {
  it(`renders correctly`, () => {
    const tree = setup().toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly', () => {
    setup();

    expect(screen.getByText(/Bem vindo/i)).toBeOnTheScreen()
    expect(screen.getByText(/de volta!/i)).toBeOnTheScreen()
    expect(screen.getByText(/sign in/i)).toBeOnTheScreen()
    expect(screen.getByText(/sign up/i)).toBeOnTheScreen()
    expect(screen.getByTestId('email-input')).toBeOnTheScreen()
    expect(screen.getByTestId('password-input')).toBeOnTheScreen()
  })

  it('should navigate to Sign Up screen', () => {
    setup();

    const signUpButton = screen.getByText(/sign up/i)
    fireEvent.press(signUpButton)
    expect(screen.getByText(/card aqui/i)).toBeOnTheScreen()
  })

  describe('when login button is pressed', () => {
    describe('and email and password is filled', () => {
      let authServiceSpy: any;

      beforeEach(() => {
        authServiceSpy = jest.spyOn(authService, 'auth')
        setup();
        const email = screen.getByTestId('email-input')
        const password = screen.getByTestId('password-input')

        // fireEvent.changeText(email, 'email@email.com')
        // fireEvent.changeText(password, 'secret')
        fireEvent.changeText(email, 'user@email.com')
        fireEvent.changeText(password, '123456789')

        const signInButton = screen.getByText(/sign in/i)
        fireEvent.press(signInButton)
      })

      it('should navigate to Dashboard Page', async () => {
        const expectedProps = {
          "user": {
            "email": "user@email.com",
            "password": "123456789"
          }
        }

        expect(authServiceSpy).toBeCalledWith(expectedProps)
      })
    })
  })
})
