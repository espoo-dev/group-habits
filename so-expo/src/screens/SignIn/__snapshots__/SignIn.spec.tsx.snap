// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src > screens > SignIn renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="SignIn"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            onLayout={[Function]}
            style={
              [
                [
                  {
                    "backgroundColor": "#FBFBFC",
                  },
                ],
                {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "position": "relative",
                  },
                  {
                    "height": "100%",
                  },
                  {
                    "width": "100%",
                  },
                  {
                    "alignItems": "center",
                  },
                  {
                    "justifyContent": "space-between",
                  },
                  {
                    "paddingBottom": 32,
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "display": "flex",
                    },
                    {
                      "justifyContent": "space-between",
                    },
                    {
                      "height": "100%",
                    },
                    {
                      "width": "100%",
                    },
                    {
                      "paddingLeft": 24,
                      "paddingRight": 24,
                    },
                    {
                      "paddingBottom": 96,
                      "paddingTop": 96,
                    },
                    {
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 0,
                    },
                    {
                      "position": "absolute",
                    },
                    {
                      "zIndex": 10,
                    },
                    {
                      "display": "flex",
                    },
                    {
                      "alignItems": "flex-start",
                    },
                    {
                      "justifyContent": "flex-start",
                    },
                    {
                      "height": "20%",
                    },
                    {
                      "backgroundColor": "#1A54F8",
                    },
                    {
                      "paddingTop": 8,
                    },
                    {
                      "paddingLeft": 24,
                      "paddingRight": 24,
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#FBFBFC",
                      },
                      {
                        "fontWeight": "700",
                      },
                      {
                        "fontSize": 30,
                        "lineHeight": 36,
                      },
                      {
                        "marginTop": 8,
                      },
                    ]
                  }
                >
                  Bem vindo
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#FBFBFC",
                      },
                      {
                        "fontWeight": "700",
                      },
                      {
                        "fontSize": 24,
                        "lineHeight": 32,
                      },
                      {
                        "backgroundColor": "#374151",
                      },
                      {
                        "paddingBottom": 8,
                        "paddingLeft": 8,
                        "paddingRight": 8,
                        "paddingTop": 8,
                      },
                    ]
                  }
                >
                  de volta!
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "display": "flex",
                    },
                    {
                      "justifyContent": "space-between",
                    },
                    {
                      "height": "100%",
                    },
                    {
                      "width": "100%",
                    },
                    {
                      "paddingLeft": 24,
                      "paddingRight": 24,
                    },
                    {
                      "paddingBottom": 96,
                      "paddingTop": 96,
                    },
                    {
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 0,
                    },
                    {
                      "width": "100%",
                    },
                    {
                      "height": "100%",
                    },
                    {
                      "display": "flex",
                    },
                    {
                      "alignItems": "center",
                    },
                    {
                      "backgroundColor": "#FBFBFC",
                    },
                    {
                      "paddingLeft": 24,
                      "paddingRight": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    [
                      {
                        "display": "flex",
                      },
                      {
                        "alignItems": "center",
                      },
                      {
                        "justifyContent": "flex-end",
                      },
                      {
                        "width": "100%",
                      },
                      {
                        "height": "100%",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "width": "100%",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "fontSize": 14,
                            "lineHeight": 20,
                          },
                          {
                            "fontWeight": "500",
                          },
                          {
                            "color": "#374151",
                          },
                        ]
                      }
                    >
                      Email
                    </Text>
                    <View
                      style={
                        [
                          {
                            "width": "100%",
                          },
                          {
                            "marginTop": 4,
                          },
                        ]
                      }
                    >
                      <TextInput
                        keyboardType="email-address"
                        onBlur={[Function]}
                        onChangeText={[Function]}
                        onFocus={[Function]}
                        placeholder="minha-empresa@example.com"
                        placeholderTextColor="gray"
                        style={
                          [
                            {
                              "width": "100%",
                            },
                            {
                              "paddingLeft": 16,
                            },
                            {
                              "height": 40,
                            },
                            {
                              "borderBottomLeftRadius": 6,
                              "borderBottomRightRadius": 6,
                              "borderTopLeftRadius": 6,
                              "borderTopRightRadius": 6,
                            },
                            {
                              "borderBottomWidth": 1,
                              "borderLeftWidth": 1,
                              "borderRightWidth": 1,
                              "borderTopWidth": 1,
                            },
                            {
                              "borderBottomColor": "#d1d5db",
                              "borderLeftColor": "#d1d5db",
                              "borderRightColor": "#d1d5db",
                              "borderTopColor": "#d1d5db",
                            },
                            {
                              "shadowColor": "rgba(0, 0, 0, 0.1)",
                              "shadowOffset": {
                                "height": 1,
                                "width": 0,
                              },
                              "shadowOpacity": 1,
                              "shadowRadius": 2,
                            },
                            {
                              "fontSize": 14,
                              "lineHeight": 20,
                            },
                          ]
                        }
                        testID="email-input"
                        value=""
                      />
                    </View>
                  </View>
                  <View
                    style={
                      [
                        {
                          "width": "100%",
                        },
                        {
                          "marginTop": 8,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "fontSize": 14,
                            "lineHeight": 20,
                          },
                          {
                            "fontWeight": "500",
                          },
                          {
                            "color": "#374151",
                          },
                        ]
                      }
                    >
                      Senha
                    </Text>
                    <View
                      style={
                        [
                          {
                            "width": "100%",
                          },
                          {
                            "marginTop": 4,
                          },
                        ]
                      }
                    >
                      <TextInput
                        keyboardType="default"
                        onBlur={[Function]}
                        onChangeText={[Function]}
                        onFocus={[Function]}
                        placeholder="****************"
                        placeholderTextColor="gray"
                        secureTextEntry={true}
                        style={
                          [
                            {
                              "width": "100%",
                            },
                            {
                              "paddingLeft": 16,
                            },
                            {
                              "height": 40,
                            },
                            {
                              "borderBottomLeftRadius": 6,
                              "borderBottomRightRadius": 6,
                              "borderTopLeftRadius": 6,
                              "borderTopRightRadius": 6,
                            },
                            {
                              "borderBottomWidth": 1,
                              "borderLeftWidth": 1,
                              "borderRightWidth": 1,
                              "borderTopWidth": 1,
                            },
                            {
                              "borderBottomColor": "#d1d5db",
                              "borderLeftColor": "#d1d5db",
                              "borderRightColor": "#d1d5db",
                              "borderTopColor": "#d1d5db",
                            },
                            {
                              "shadowColor": "rgba(0, 0, 0, 0.1)",
                              "shadowOffset": {
                                "height": 1,
                                "width": 0,
                              },
                              "shadowOpacity": 1,
                              "shadowRadius": 2,
                            },
                            {
                              "fontSize": 14,
                              "lineHeight": 20,
                            },
                          ]
                        }
                        testID="password-input"
                        value=""
                      />
                    </View>
                  </View>
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "alignItems": "center",
                        },
                        {
                          "borderBottomWidth": 1,
                          "borderLeftWidth": 1,
                          "borderRightWidth": 1,
                          "borderTopWidth": 1,
                        },
                        {
                          "shadowColor": "rgba(0, 0, 0, 0.1)",
                          "shadowOffset": {
                            "height": 1,
                            "width": 0,
                          },
                          "shadowOpacity": 1,
                          "shadowRadius": 2,
                        },
                        {
                          "borderBottomLeftRadius": 6,
                          "borderBottomRightRadius": 6,
                          "borderTopLeftRadius": 6,
                          "borderTopRightRadius": 6,
                        },
                        {
                          "paddingLeft": 24,
                          "paddingRight": 24,
                        },
                        {
                          "paddingBottom": 12,
                          "paddingTop": 12,
                        },
                        {
                          "borderBottomColor": "transparent",
                          "borderLeftColor": "transparent",
                          "borderRightColor": "transparent",
                          "borderTopColor": "transparent",
                        },
                        {
                          "backgroundColor": "#4f46e5",
                        },
                        {
                          "width": "100%",
                        },
                        {
                          "marginTop": 12,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "display": "flex",
                          },
                          {
                            "textAlign": "center",
                          },
                          {
                            "width": "100%",
                          },
                          {
                            "fontWeight": "700",
                          },
                          {
                            "color": "#fff",
                          },
                        ]
                      }
                    >
                      Sign in
                    </Text>
                  </View>
                  <View
                    style={
                      [
                        {
                          "position": "relative",
                        },
                        {
                          "display": "flex",
                        },
                        {
                          "alignItems": "center",
                        },
                        {
                          "justifyContent": "center",
                        },
                        {
                          "width": "100%",
                        },
                        {
                          "marginTop": 8,
                        },
                      ]
                    }
                  >
                    <View
                      aria-hidden={true}
                      style={
                        [
                          {
                            "position": "absolute",
                          },
                          {
                            "top": 0,
                          },
                          {
                            "right": 0,
                          },
                          {
                            "bottom": 0,
                          },
                          {
                            "left": 0,
                          },
                          {
                            "display": "flex",
                          },
                          {
                            "alignItems": "center",
                          },
                          {
                            "justifyContent": "center",
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "width": "100%",
                            },
                            {
                              "borderTopWidth": 1,
                            },
                            {
                              "borderBottomColor": "#d1d5db",
                              "borderLeftColor": "#d1d5db",
                              "borderRightColor": "#d1d5db",
                              "borderTopColor": "#d1d5db",
                            },
                          ]
                        }
                      />
                    </View>
                    <View
                      style={
                        [
                          {
                            "position": "relative",
                          },
                          {
                            "display": "flex",
                          },
                          {
                            "alignItems": "center",
                          },
                          {
                            "justifyContent": "center",
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "backgroundColor": "#fff",
                            },
                            {
                              "paddingLeft": 8,
                              "paddingRight": 8,
                            },
                            {
                              "fontSize": 14,
                              "lineHeight": 20,
                            },
                            {
                              "color": "#6b7280",
                            },
                          ]
                        }
                      >
                        Ou
                      </Text>
                    </View>
                  </View>
                  <View
                    style={
                      [
                        {
                          "width": "100%",
                        },
                        {
                          "marginTop": 8,
                        },
                      ]
                    }
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        [
                          {
                            "alignItems": "center",
                          },
                          {
                            "borderBottomWidth": 1,
                            "borderLeftWidth": 1,
                            "borderRightWidth": 1,
                            "borderTopWidth": 1,
                          },
                          {
                            "shadowColor": "rgba(0, 0, 0, 0.1)",
                            "shadowOffset": {
                              "height": 1,
                              "width": 0,
                            },
                            "shadowOpacity": 1,
                            "shadowRadius": 2,
                          },
                          {
                            "borderBottomLeftRadius": 6,
                            "borderBottomRightRadius": 6,
                            "borderTopLeftRadius": 6,
                            "borderTopRightRadius": 6,
                          },
                          {
                            "paddingLeft": 24,
                            "paddingRight": 24,
                          },
                          {
                            "paddingBottom": 12,
                            "paddingTop": 12,
                          },
                          {
                            "borderBottomColor": "transparent",
                            "borderLeftColor": "transparent",
                            "borderRightColor": "transparent",
                            "borderTopColor": "transparent",
                          },
                          {
                            "backgroundColor": "#e0e7ff",
                          },
                          {
                            "width": "100%",
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "display": "flex",
                            },
                            {
                              "textAlign": "center",
                            },
                            {
                              "width": "100%",
                            },
                            {
                              "fontWeight": "700",
                            },
                            {
                              "color": "#4338ca",
                            },
                          ]
                        }
                      >
                        Sign up
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
